src/parser/RegExpParser.mts

```diff
@@ -927,4 +1169,247 @@ export class RegExpParser {
+        },
+      };
+    }
+    if (this.eat('\\')) {
+      // ClassSetOperand :: ClassStringDisjunction :: \q{ ClassStringDisjunctionContents }
+      if (this.eat('q')) {
+        this.expect('{');
+        const ClassStringDisjunction = this.parseClassStringDisjunctionContents();
+        this.expect('}');
+        return {
+          type: 'ClassSetOperand',
+          production: 'ClassStringDisjunction',
+          ClassStringDisjunction,
+        };
+      }
+      // ClassSetOperand :: NestedClass :: \ CharacterClassEscape[+UnicodeMode]
+      const escape = this.scope(
+        { UnicodeMode: true },
+        () => this.maybeParseCharacterClassEscape(),
+      );
+      if (!escape) {
+        this.raise(`Expect a CharacterClassEscape but ${this.peek()}`);
+      }
+      return {
+        type: 'ClassSetOperand',
+        production: 'NestedClass',
+        NestedClass: { type: 'NestedClass', production: 'CharacterClassEscape', CharacterClassEscape: escape },
+      };
+    }
+    const ClassSetCharacter = this.maybeParseClassSetCharacter();
+    if (!ClassSetCharacter) {
+      this.raise(`Unexpected ${this.peek()}`);
+    }
+    return { type: 'ClassSetOperand', production: 'ClassSetCharacter', ClassSetCharacter };
+  }
+
+  //  ClassSetCharacter ::
+  //    [lookahead ∉ ClassSetReservedDoublePunctuator] SourceCharacter but not ClassSetSyntaxCharacter
+  //    \ CharacterEscape[+UnicodeMode]
+  //    \ ClassSetReservedPunctuator
+  //    \b
+  private maybeParseClassSetCharacter(): ParseNode.RegExp.ClassSetCharacter | undefined {
+    Assert(this.inUnicodeSetMode);
+    const nextTwo = this.peek(2);
+    // ClassSetCharacter :: \b
+    if (nextTwo === '\\b') {
+      this.position += 2;
+      return { type: 'ClassSetCharacter', production: 'UnicodeCharacter', UnicodeCharacter: '\\b' as UnicodeCharacter };
+    }
+
+    // ClassSetCharacter :: [lookahead ∉ ClassSetReservedDoublePunctuator] SourceCharacter but not ClassSetSyntaxCharacter
+    if (
+      // [lookahead ∉ ClassSetReservedDoublePunctuator]
+      !'&& !! ## $$ %% ** ++ ,, .. :: ;; << == >> ?? @@ ^^ `` ~~'.split(' ').includes(nextTwo)
+      // and not ClassSetSyntaxCharacter
+      && !'( ) [ ] { } / - \\ |'.split(' ').includes(nextTwo[0])
+    ) {
+      // parse SourceCharacter
+      return { type: 'ClassSetCharacter', production: 'UnicodeCharacter', UnicodeCharacter: this.parseSourceCharacter() as UnicodeCharacter };
+    }
+
+    // all production left requires a \ at the beginning
+    if (nextTwo[0] !== '\\') {
+      return undefined;
+    }
+
+    // \ ClassSetReservedPunctuator
+    if ('& - ! # % , : ; < = > @ ` ~'.split(' ').includes(nextTwo[1])) {
+      this.position += 2;
+      return { type: 'ClassSetCharacter', production: 'UnicodeCharacter', UnicodeCharacter: nextTwo[1] as UnicodeCharacter };
+    }
+
+    // anything that can start a Character Escape
+    if ('f n r t v c 0 x u / ^ $ \\ . * + ? ( ) [ ] { } |'.split(' ').includes(nextTwo[1])) {
```

Comment:
I know engine262 is more concerned with correctness over efficiency.  But this function seems like it would benefit from having these long split strings be constants defined outside the method, say, as private static members.